### DISCUSSION TITLE
incsearch first character always match from beginning

### DISCUSSION THREAD

**Discussion by gary-ruizhang** [JetBrains/ideavim#663] [General]
<img width="866" alt="CleanShot 2023-07-11 at 16 00 16@2x" src="https://github.com/JetBrains/ideavim/assets/32344731/4e1cb416-e716-4a2b-abb9-7e7e67cb17dd">
I'm at line 349 and I want to search "Cust", I press / and enter the first char 'C', match jump to the beginning of the file.
<img width="1488" alt="CleanShot 2023-07-11 at 16 04 36@2x" src="https://github.com/JetBrains/ideavim/assets/32344731/9078ff74-875c-4d5a-b541-dd476e7380f3">
look at the right corner of status line, I'm still on the line 349. After I enter the second char, match back to the correct position. This issue just happened for the first character.

**Comment by citizenmatt** [comment]
Could you share your `~/.ideavimrc` please? I don't see the same behaviour - when I start searching, it highlights the next match below the text cursor, even if there are more matches above it.

**Reply by gary-ruizhang** [reply]
```
let mapleader = " "

set surround
set commentary
set ignorecase
set smartcase
set incsearch              " Highlight while searching with / or ?.
set hlsearch               " Keep matches highlighted.
set visualbell
set noerrorbells
set keep-english-in-normal

nnoremap <silent> <C-l> :<C-u>nohl<CR><C-l>
nnoremap g; g;zz
nnoremap g, g,zz
nnoremap ' `
nnoremap ` '
nnoremap * *zzzv
nnoremap # #zzzv
nnoremap n nzzzv
nnoremap N Nzzzv
nnoremap ; :
vnoremap ; :
nnoremap : ;
vnoremap : ;
nnoremap j gj
nnoremap k gk

nnoremap gr gd
nnoremap gn :action GotoNextError<CR>
nnoremap gp :action GotoPreviousError<CR>
nnoremap g[ :action GotoImplementation<CR>
map ga <Action>(ShowIntentionActions)
nmap <C-o> :action Back<CR>
nmap <c-i> :action Forward<CR>


" Press `f` to activate AceJump
nmap f <Action>(AceAction)
" Press `F` to activate Target Mode
nmap F <Action>(AceTargetAction)
" Press `S` in normal mode to activate AceJump mode before the caret
nmap S <Action>(AceBackwardAction)
" Press `s` in normal mode to activate AceJump mode after the caret
nmap s <Action>(AceForwardAction)

```
